app/src/main/java/com/celzero/bravedns/util/Utilities.kt

```diff
@@ -199,43 +208,44 @@ class Utilities {
             return format.format(date)
         }
 
-        fun convertLongToDate(timeStamp: Long): String{
+        fun convertLongToDate(timeStamp: Long): String {
             val date = Date(timeStamp)
             val format = SimpleDateFormat("yy.MM (dd)", Locale.US)
             return format.format(date)
         }
 
-        fun prepareServersToRemove(servers: String, liveServers: String): String{
-            val serverList  = servers.split(",")
+        fun prepareServersToRemove(servers: String, liveServers: String): String {
+            val serverList = servers.split(",")
             val liveServerList = liveServers.split(",")
-            if(DEBUG) Log.d(LOG_TAG, "Servers to remove - $serverList -- $liveServerList")
-            var serversToSend : String =""
-            serverList.forEach{
-                if(!liveServerList.contains(it)){
+            if (DEBUG) Log.d(LOG_TAG, "Servers to remove - $serverList -- $liveServerList")
+            var serversToSend: String = ""
+            serverList.forEach {
+                if (!liveServerList.contains(it)) {
                     serversToSend += "$it,"
                 }
             }
-            if(DEBUG) Log.d(LOG_TAG, "Servers to remove - $serversToSend")
+            if (DEBUG) Log.d(LOG_TAG, "Servers to remove - $serversToSend")
             serversToSend = serversToSend.dropLast(1)
             return serversToSend
         }
 
-        fun showToastInMidLayout(context: Context?, message: String, toastLength: Int){
+        fun showToastInMidLayout(context: Context?, message: String, toastLength: Int) {
             try {
                 val toast = Toast.makeText(context, message, toastLength)
                 toast.setGravity(Gravity.CENTER, 0, 0)
                 toast.show()
-            }catch (e: java.lang.IllegalStateException){
+            } catch (e: java.lang.IllegalStateException) {
                 Log.w(LOG_TAG, "Show Toast issue : ${e.message}", e)
             }
         }
 
-        fun isIPLocal(ipAddress: String): Boolean{
-            return try{
-                val ip = InetAddress.getByName(ipAddress)
+        fun isIPLocal(ipAddress: String): Boolean {
+            return try {
+                // InetAddresses - 'com.google.common.net.InetAddresses' is marked unstable with @Beta
+                val ip = InetAddresses.forString(ipAddress)
                 val regex = Regex("(^127\\.0\\.0\\.1)|(^10\\.)|(^172\\.1[6-9]\\.)|(^172\\.2[0-9]\\.)|(^172\\.3[0-1]\\.)|(^192\\.168\\.)")
```

Comment:
where was this regex got from? Include a link to it.

Also, regex doesn't seem like a performant way to detect a site-local-ip? Why not use: https://developer.android.com/reference/java/net/InetAddress#isSiteLocalAddress() and https://developer.android.com/reference/java/net/InetAddress#isLoopbackAddress()

Reply:
There is a check of isAnyLocalAddress from InetAddress and added the regex along with that check. 
Included the link in comment.

Reply:
Removed the regex and using the APIs.